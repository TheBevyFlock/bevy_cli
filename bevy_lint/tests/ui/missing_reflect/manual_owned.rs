//! A version of the normal UI test that replaces `#[derive(...)]` attributes with manual
//! implementations.

// We want to require annotations for all errors, but don't need them for notes.
//@require-annotations-for-level: ERROR

#![feature(register_tool)]
#![register_tool(bevy)]
#![allow(dead_code)]
#![deny(bevy::missing_reflect)]

use bevy::{ecs::component::StorageType, prelude::*};

struct MyComponent;
//~^ ERROR: defined a component without a `Reflect` implementation
//~| HELP: `Reflect` can be automatically derived

//~v NOTE: `Component` implemented here
impl Component for MyComponent {
    const STORAGE_TYPE: StorageType = StorageType::Table;
}

struct MyResource;
//~^ ERROR: defined a resource without a `Reflect` implementation
//~| HELP: `Reflect` can be automatically derived

//~v NOTE: `Resource` implemented here
impl Resource for MyResource {}

struct MyEvent;
//~^ ERROR: defined an event without a `Reflect` implementation
//~| HELP: `Reflect` can be automatically derived

impl Component for MyEvent {
    const STORAGE_TYPE: StorageType = StorageType::Table;
}

//~v NOTE: `Event` implemented here
impl Event for MyEvent {}
