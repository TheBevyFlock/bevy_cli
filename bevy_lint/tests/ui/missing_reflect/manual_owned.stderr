error: defined an event without a `Reflect` implementation
  --> tests/ui/missing_reflect/manual_owned.rs:30:1
   |
30 | struct MyEvent;
   | ^^^^^^^^^^^^^^^
   |
note: `Event` implemented here
  --> tests/ui/missing_reflect/manual_owned.rs:39:1
   |
39 | impl Event for MyEvent {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
note: the lint level is defined here
  --> tests/ui/missing_reflect/manual_owned.rs:10:9
   |
10 | #![deny(bevy::missing_reflect)]
   |         ^^^^^^^^^^^^^^^^^^^^^
help: `Reflect` can be automatically derived
   |
30 + #[derive(Reflect)]
31 | struct MyEvent;
   |

error: defined a component without a `Reflect` implementation
  --> tests/ui/missing_reflect/manual_owned.rs:14:1
   |
14 | struct MyComponent;
   | ^^^^^^^^^^^^^^^^^^^
   |
note: `Component` implemented here
  --> tests/ui/missing_reflect/manual_owned.rs:19:1
   |
19 | / impl Component for MyComponent {
20 | |     const STORAGE_TYPE: StorageType = StorageType::Table;
21 | | }
   | |_^
help: `Reflect` can be automatically derived
   |
14 + #[derive(Reflect)]
15 | struct MyComponent;
   |

error: defined a resource without a `Reflect` implementation
  --> tests/ui/missing_reflect/manual_owned.rs:23:1
   |
23 | struct MyResource;
   | ^^^^^^^^^^^^^^^^^^
   |
note: `Resource` implemented here
  --> tests/ui/missing_reflect/manual_owned.rs:28:1
   |
28 | impl Resource for MyResource {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: `Reflect` can be automatically derived
   |
23 + #[derive(Reflect)]
24 | struct MyResource;
   |

error: aborting due to 3 previous errors

